tools/check_version.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python
+
+import sys
+from typing import List, Callable
+from ctypes import c_uint8, c_uint16
+import argparse
+import re
+
+#
+# MARK: - Standards
+#
+
+class Standard:
+    def __init__(self, is_version_compliant:Callable[[str], bool], success_message:str, error_message:str):
+        self.is_version_compliant:Callable[[str], bool] = is_version_compliant
+        self.success_message:str = success_message
+        self.error_message:str = error_message
+
+# Source: https://semver.org/
+semver_regex = "^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$"
+semver_standard = Standard(is_version_compliant=lambda version: re.match(semver_regex, version) is not(None),
+                           success_message="✅ Version is compliant with Semantic Versioning 2.0.0.",
+                           error_message="❌ Version is not compliant with Semantic Versioning 2.0.0.")
+
+
+def mcuboot_is_version_compliant(version: str):
+    version_splited = version.split(sep='.')
+    if len(version_splited) != 3:
+        return False
+    try:
+        if not(int(version_splited[0]) <= c_uint8(-1).value):
+            return False
+        if not(int(version_splited[1]) <= c_uint8(-1).value):
+            return False
+        if not(int(version_splited[2]) <= c_uint16(-1).value):
+            return False
+    except:
+        return False
+    return True
+
+mcuboot_standard = Standard(is_version_compliant=mcuboot_is_version_compliant,
+                            success_message="✅ Version is compliant with MCUboot.",
+                            error_message="❌ Version is not compliant with MCUboot.")
+
+list_of_standards = {"semver": semver_standard,
+                     "mcuboot": mcuboot_standard}
+
+#
+# MARK: - argparse
+#
+
+parser = argparse.ArgumentParser(description='Check version is valid with standard')
+
+parser.add_argument('input_version', metavar='input_version', type=str,
```

Comment:
I would pass the file by default, instead of the version

this way we can use the script in CI to make sure that version bumps are compliant.

`./tools/check_os_version.py ./config/leka_os_version`